app/src/main/java/org/sagebase/crf/step/CrfInstructionStepLayout.java

```diff
@@ -0,0 +1,138 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+package org.sagebase.crf.step;
+
+import android.content.Context;
+import android.support.v4.content.res.ResourcesCompat;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.Button;
+
+import org.researchstack.backbone.result.StepResult;
+import org.researchstack.backbone.step.Step;
+import org.researchstack.backbone.ui.callbacks.StepCallbacks;
+import org.researchstack.backbone.ui.step.layout.InstructionStepLayout;
+
+import org.sagebionetworks.research.crf.R;
+
+/**
+ * Created by TheMDP on 10/24/17.
+ */
+
+public class CrfInstructionStepLayout extends InstructionStepLayout {
+
+    private CrfInstructionStep crfStep;
+
+    private Button nextButton;
+
+    public CrfInstructionStepLayout(Context context) {
+        super(context);
+    }
+
+    public CrfInstructionStepLayout(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public CrfInstructionStepLayout(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+    public CrfInstructionStepLayout(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
+        super(context, attrs, defStyleAttr, defStyleRes);
+    }
+
+    @Override
+    public int getContentResourceId() {
+        return R.layout.crf_step_layout_instruction;
+    }
+
+    @Override
+    public void initialize(Step step, StepResult result) {
+        validateAndSetCrfStep(step);
+        super.initialize(step, result);
+    }
+
+    protected void validateAndSetCrfStep(Step step) {
+        if (!(step instanceof CrfInstructionStep)) {
+            throw new IllegalStateException("CrfInstructionStepLayout only works with CrfInstructionStep");
+        }
+        this.crfStep = (CrfInstructionStep)step;
+    }
+
+    @Override
+    public int getContentContainerLayoutId() {
+        return R.id.crf_step_layout_container;
+    }
+
+    @Override
+    public int getFixedSubmitBarLayoutId() {
+        return R.layout.crf_step_layout_container;
+    }
+
+    @Override
+    public void connectStepUi(int titleRId, int textRId, int imageRId, int detailRId) {
+        super.connectStepUi(
+                R.id.crf_intruction_title,
+                R.id.crf_intruction_text,
+                R.id.crf_image_view,
+                R.id.crf_instruction_more_detail_text);
+
+        nextButton = findViewById(R.id.button_go_forward);
+    }
+
+    @Override
+    public void refreshStep() {
+        super.refreshStep();
+
+        if (crfStep.buttonText != null) {
+            nextButton.setText(crfStep.buttonText);
+        }
+        if (crfStep.buttonType != null) {
+            switch (crfStep.buttonType) {
+                case DEFAULT:
```

Comment:
Could these be defined on the enum?

    enum CRFStepButtonType {
        @ColorRes int getTextColor();
        @DrawableRes int getBackground();
    }

Reply:
That's a good idea.  I'm still waiting on the assets to be delivered from Woody, so I can't really mess around with displaying those yet.  I'll definitely keep that in mind when we start tackling the tasks.